presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -607,6 +607,10 @@ private synchronized void checkLockHeld(String message)
 
     private static ListenableFuture<?> firstFinishedFuture(List<ListenableFuture<?>> futures)
     {
+        if (futures.size() == 1) {
```

Comment:
This function appears like duplicate of `whenAnyComplete` in airlift `MoreFutures`. What about using that instead?